mongo.y

```diff
@@ -108,7 +124,9 @@ lexer ('\'':cs) = lexString cs '\''
 lexer ('\"':cs) = lexString cs '\"' 
 lexer (c:cs) 
     | isSpace c = lexer cs
-    | isDigit c = lexNum (c:cs)
+    | isDigit c || c == '-' = case lexNum (c:cs) of
+	Just (n, rest) -> n : lexer rest
```

Comment:
This is really nitpicky, but can we make it `t` instead of `n` so that it's clear it's a token?